src/services/ledger/index.ts

```diff
@@ -370,23 +369,21 @@ export const LedgerService = (): ILedgerService => {
         type: LedgerTransactionType.Invoice,
         pending: false,
         hash: paymentHash,
-        fee: feeLightningLiquidity,
-        feeUsd: usdFeeLightningLiquidity,
+        fee: feeInboundLiquidity,
+        feeUsd: usdFeeInboundLiquidity,
         sats,
         usd: usdDisplay,
-        currency,
+        currency: WalletCurrency.Btc,
       }
 
-      fiat // TODO handle the case where currency = USD
-
       const entry = MainBook.entry(description)
       entry
-        .credit(liabilitiesWalletId, sats - feeLightningLiquidity, metadata)
+        .credit(liabilitiesWalletId, sats - usdFeeInboundLiquidity, metadata)
```

Comment:
This shouldn't type check (don't know how to enforce). Subtracting `Satoshi - FiatAmount` seems like its clearly an error.

Reply:
seems because they are both number that is working 😒

maybe an option would be to have a .add() to those objects (for everything related to Satoshis/Fiat), and by convention not use the + - / any more.

Reply:
I think we should have some generic calculation functions:
```
const add = <T extends number> (a: T, b:T) => {
  return a + b
}
const sub = <T extends number> (a: T, b:T) => {
  return a - b
}
```
 etc.
 
see https://stackoverflow.com/questions/54470329/how-to-make-a-generic-add-operator-in-typescript-that-works-with-numbers-and-str